### ISSUE TITLE
Actors behaviour

### ISSUE THREAD

**Issue by dilumnavanjana** [dilumnavanjana/celluloid#43]
Here is my latest changes for Unlocker https://github.com/dilumnavanjana/celluloid/commit/ca845e72f91384401df839e3ef983254fd6f4558

Then I tested some simple Test implementations & deadlock happening implementations.

With **ruby**
- Most of the deadlock happening samples crashes because of the deadlock.
- When inspecting the actors of simple Test class, there is one thread that runs always & others are in sleeping status.

With **jruby**
- When inspecting the actors of all the classes including deadlock & simple test, all the thread are in the sleeping status all the time.

With **rbx**
- Results are same as **ruby**

@digitalextremist Do you have any idea why all the thread are in the sleeping status with `jruby`?
And why there is only one thread is in the running status though there are more than one `celluloid` `async` processes going on?

**Comment by digitalextremist** [comment]
By the way, I've been seeing you using timers in `Toggl`. Nice! Keep it up.

What versions of each RVM are you using?

Keep in mind, method calls are tasks, and tasks are fibers, unless you are using the threaded task class. You might want to test that way by setting the environment variable accordingly.

Speaking of, it would be nice if you could write a piece of documentation in the Celluloid Wiki about how to switch between the three active task handlers. Right now we have Fiber pools, Fibers, and Threads.

**Comment by digitalextremist** [comment]
When you check the actor, I would not use `inspect` because that could get caught up in a lock. I suggest you place a testing method on the actor yourself, and call that... making sure it cannot be locked by that call. Otherwise, research and report on how actors are structured. I would be interested to read your interpretation of how an actor behaves, from instantiation to death.

**Comment by dilumnavanjana** [comment]
I'm using `ruby-2.1.1`, `jruby-1.7.19` & `rbx-2.5.2`

Are Method calls inside the library itself too tasks?Have to read some more about celluloid Fiber. I'll start writing a documentation for active task handler once finish reading on Celluloid Fiber.

So should I use `@mutex.synchronize` or `@mutex.lock` to avoid locking in the testing method?

**Comment by dilumnavanjana** [comment]
Here I wrote [Fiber pools, Fibers, and Threads](https://github.com/dilumnavanjana/celluloid/wiki/Fiber-pools,-Fibers,-and-Threads) wiki inside my forked project. Need your suggestions & feedback to improve it more.

**Comment by digitalextremist** [comment]
That's a very good start! Did you gain any new insights into the project doing that?

**Comment by dilumnavanjana** [comment]
Yes I got some knowledge about how Fiber & Fiber pools works by reading for this. It'll help with the implementation of Unlocker.
